scenes/map/map.gd

```diff
@@ -0,0 +1,36 @@
+extends Node2D
+
+@onready var ship = %ShipSprite
+@onready var island_0: Sprite2D = %Island0Sprite
+@onready var island_1: Sprite2D = %Island1Sprite
+@onready var island_2: Sprite2D = %Island2Sprite
+
+var goal_island = null
```

Comment:
Instead of having an id, we can keep a reference to the node instead, so we wouldn't need to get it every frame